src/libCUserNotification/libCUserNotification.c

```diff
@@ -59,14 +84,21 @@ void set_sound_name(usernotification_t *notif, char * sound_name) {
 
 void post_notification(usernotification_t *notif) {
     id pool = (id)objc_getClass("NSAutoreleasePool");
-    
-    pool = objc_msgSend(pool,
-                        sel_registerName("alloc"),
-                        sel_registerName("init"));
+    pool = objc_msgSend(pool, sel_registerName("alloc"), sel_registerName("init"));
+
+    set_default_bundle_id();
+    id def_notif_center = get_default_user_notif_center();
 
-    set_bundle_id();
-    objc_msgSend(get_default_user_notif_center(), sel_registerName("deliverNotification:"), notif->notification);
+    if(notif->bundleID) {
+        CFStringRef cf_bid = c_cfstr(notif->bundleID);
+        set_custom_bundle_id(def_notif_center, cf_bid);
+        objc_msgSend(def_notif_center, sel_registerName("deliverNotification:"), notif->notification);
+        sleep(1);
+        CFRelease(cf_bid);
+    } else {
```

Comment:
The reason I had to set it up like this is because `cf_bid` could not be freed before the call to `sleep`.

Reply:
Looks good, seems like the cleanest/best way to do it.